pkg/services/hivelocity/device/device.go

```diff
@@ -397,9 +397,9 @@ func (s *Service) actionDeviceProvisioned(ctx context.Context) actionResult {
 			infrav1.DeviceReadyCondition,
 			infrav1.DeviceTagsInvalidReason,
 			clusterv1.ConditionSeverityError,
-			fmt.Sprintf("device %d has invalid tags", device.DeviceId),
+			fmt.Sprintf("device %d has invalid tags: %s", device.DeviceId, err.Error()),
 		)
-		record.Warnf(s.scope.HivelocityMachine, "DeviceTagsInvalid", "Hivelocity device not found.")
+		record.Warnf(s.scope.HivelocityMachine, "DeviceTagsInvalid", fmt.Sprintf("Hivelocity device not found: %s", err.Error()))
```

Comment:
device not found doesn't fit AFAIK

Reply:
@janiskemper yes, you are right. Yesterday I had this error since my machines got stolen by someone else. I updated it, so that the user gets a better err message.

Reply:
@janiskemper the new code creates a better error message.